File: app/src/main/java/bj/drunkmessengerblocker/blacklist/BlacklistComponent.java
package bj.drunkmessengerblocker.blacklist;

import bj.drunkmessengerblocker.ActivityScope;
import bj.drunkmessengerblocker.AppComponent;
import dagger.Component;

/**
 * Created by j on 16/01/2017.
 */
@ActivityScope
@Component(dependencies = {AppComponent.class}, modules = {BlacklistModule.class})
public interface BlacklistComponent
{
    void inject(BlacklistActivity activity);
}

File: app/src/main/java/bj/drunkmessengerblocker/blacklist/BlacklistComponent.kt
package bj.drunkmessengerblocker.blacklist

import bj.drunkmessengerblocker.ActivityScope
import bj.drunkmessengerblocker.AppComponent
import dagger.Component

/**
 * Created by Josh Laird on 23/05/2017.
 */
@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(BlacklistModule::class))
interface BlacklistComponent {
    fun inject(activity: BlacklistActivity)
}